txnsync/txngroups.go

```diff
@@ -99,28 +99,32 @@ func (s *syncState) encodeTransactionGroups(inTxnGroups []transactions.SignedTxG
 	}, nil
 }
 
-func (s *syncState) compressTransactionGroupsBytes(data []byte) ([]byte, byte) {
+func (s *syncState) compressTransactionGroupsBytes(uncompressedData []byte) ([]byte, byte) {
 	b := getMessageBuffer()
-	if cap(b) < len(data) {
+	if cap(b) < len(uncompressedData) {
 		releaseMessageBuffer(b)
-		b = make([]byte, 0, len(data))
+		b = make([]byte, 0, len(uncompressedData))
 	}
-	_, out, err := compress.Compress(data, b, 1)
+	_, compressedData, err := compress.Compress(uncompressedData, b, 1)
 	if err != nil {
 		if errors.Is(err, compress.ErrShortBuffer) {
-			s.log.Debugf("compression had negative effect, made message bigger: original msg length: %d", len(data))
+			s.log.Debugf("compression had negative effect, made message bigger: original msg length: %d", len(uncompressedData))
 		} else {
-			s.log.Warnf("failed to compress %d bytes txnsync msg: %v", len(data), err)
+			s.log.Warnf("failed to compress %d bytes txnsync msg: %v", len(uncompressedData), err)
 		}
 		releaseMessageBuffer(b)
-		return data, compressionFormatNone
+		return uncompressedData, compressionFormatNone
 	}
-	if len(data) > len(out)*maxCompressionRatio {
-		s.log.Infof("compression exceeded compression ratio: compressed data len: %d", len(out))
+	if len(uncompressedData) > len(compressedData)*maxCompressionRatio {
+		s.log.Infof("compression exceeded compression ratio: compressed data len: %d", len(compressedData))
 		releaseMessageBuffer(b)
-		return data, compressionFormatNone
+		return uncompressedData, compressionFormatNone
+	} else if len(uncompressedData) <= len(compressedData) {
+		// compression wasn't effective and ended up spending more data.
```

Comment:
In theory this should never be reached. The function compress.Compress should return an error if the compression makes the compression bigger. See line 111.

Reply:
That would not happen if the buffer passed to `compress.Compress` is bigger than `len(uncompressedData)`

Reply:
Ah, I see, since we first attempt to use getMessageBuffer.